### ISSUE TITLE
MIDI file to melody

### ISSUE THREAD

**Issue by kily-wmz** [pjlab-songcomposer/songcomposer#8]
Hi！
I would like to ask if you have written a function that can convert MIDI files into the tuple notation format used in your paper. I want to process a large number of MIDI files using your model to obtain the notation. If such a function exists, could you please share it? Thank you very much.

**Comment by uu95** [comment]
@kily-wmz Hi, were you able to do it? If so, could you share the details? If not, can someone guide me on how to convert a MIDI file to the required notation?

**Comment by kily-wmz** [comment]
@uu95 Yep! I've already resolved this issue. If you need the detailed code, please leave your email address.

**Comment by uu95** [comment]
@kily-wmz please share the code at: ubaid95@yu.ac.kr. Thank you

**Comment by uu95** [comment]
@kily-wmz I wanted to confirm if you have already shared the code because I may have missed it.

Additionally, I tried to implement this on my own but can't seem to figure out why it always outputs in Chinese. Is there something I am missing here?

**Comment by kily-wmz** [comment]
@uu95 hi, I have sent the code to your email again, please check if you have received it.

**Comment by uu95** [comment]
@kily-wmz Sorry for the delay. Yes, thank you, I received it and tried running it, but I can't figure out how to control the output language of the model as I need English only. I searched the [Hugging Face multilingual documentation](https://huggingface.co/docs/transformers/en/multilingual) but still couldn't solve it. Is it related to the InternLM2 model, the tokenizer, or the format you used?

**Comment by Mark12Ding** [comment]
Thank you for your interest in our work. The language-specific prompt functionality is included in the newest version of our model. We will be releasing it soon, so please stay tuned.

Best,
Shuangrui

**Comment by youknowwyz** [comment]
@kily-wmz  Could you please share the code at: rylynn@pku.edu.cn? Thank you!

**Comment by ywdong** [comment]
@kily-wmz  Could you please share the code at: ywdong8809@qq.com Thank you!

**Comment by godofecht** [comment]
I can add this functionality to song composer, if you'd like

**Comment by PbRQianJiang** [comment]
Could you please share the code at: [202316187@stu.neu.edu.cn](mailto:202316187@stu.neu.edu.cn) Thank you!

**Comment by godofecht** [comment]
I'm just going to submit a PR for songcomposer for this... Don't get why everybody is giving out their emails like this

**Comment by blueocean0724** [comment]
Could you please share the code at: russell.johnson.navy@gmail.com. Thank you!

**Comment by Dinxin** [comment]
Could you please share the code at: 1774885528a@gmail.com. Thank you!

**Comment by PbRQianJiang** [comment]
我那个代码是我大二第一次科研写的，估计可能很多问题，很垃圾，你还要复现吗
---- Replied Message ----
From ***@***.***> Date 11/04/2025 20:50 To ***@***.***> Cc ***@***.***>***@***.***> Subject Re: [pjlab-songcomposer/songcomposer] MIDI file to melody (Issue #8) 
Dinxin left a comment (pjlab-songcomposer/songcomposer#8)
Could you please share the code at: ***@***.*** Thank you!
